src/devtools/client/webconsole/utils/autocomplete.ts

```diff
@@ -0,0 +1,249 @@
+import * as babelParser from "@babel/parser";
+import * as t from "@babel/types";
+const { filter } = require("fuzzaldrin-plus");
+
+type ObjectPreviewProperty = {
+  value: ValueFront;
+  writable: boolean;
+  configurable: boolean;
+  enumerable: boolean;
+  name: string;
+  get?: any;
+  set?: any;
+};
+type ObjectPreview = {
+  containerEntries: any[];
+  getterValues: any[];
+  properties: ObjectPreviewProperty[];
+  prototypeId: any;
+  prototypeValue: ValueFront;
+};
+type ObjectFront = {
+  className: string;
+  objectId: string;
+  preview?: ObjectPreview;
+};
+type ValueFront = {
+  _pause: any;
+  _hasPrimitive: false;
+  _primitive?: boolean;
+  _isBigInt: boolean;
+  _isSymbol: boolean;
+  _object: ObjectFront;
+  _uninitialized: boolean;
+  _unavailable: boolean;
+  _elements: null;
+  _isMapEntry: null;
+};
+type Binding = {
+  name: string;
+  value: ValueFront;
+};
+type Scope = {
+  bindings: Binding[];
+  actor: string;
+  parent: Scope;
+  functionName?: string;
+  object?: ValueFront;
+  scopeKind: string;
+  type: "block" | string;
+};
+type FrameScope = {
+  scope: Scope;
+  pending: boolean;
+  originalScopesUnavailable: boolean;
+};
+
+function getMatchString(str: string) {
```

Comment:
I find this a little confusing because the component above is called a `Match`, but this is saying `normalize this string for searching purposes` right? Maybe `normalizeString` would be a better name?

Reply:
👍 yep, and added comments to make it clear what it's for.